src/help/help.go

```diff
@@ -0,0 +1,121 @@
+// +build nobootstrap
+
+// Package help prints help messages about parts of plz.
+package help
+
+import (
+	"encoding/json"
+	"fmt"
+	"sort"
+	"strings"
+
+	"gopkg.in/op/go-logging.v1"
+
+	"utils"
+)
+
+var log = logging.MustGetLogger("help")
+
+// rulesPreamble is a message we print out before help for any built-in build rules.
+const rulesPreamble = `
+%s is a built-in build rule in Please. Instructions for use & its arguments:
+
+`
+const configPreamble = `
+%s is a config setting defined in the .plzconfig file. See "plz help plzconfig" for more information.
+
+`
+const miscPreamble = `
+%s is a general Please concept.
+
+`
+const defaultHelpMessage = `
+Please is a high-performance language-agnostic build system.
+
+Try plz help <topic> for help on a specific topic;
+plz --help if you want information on flags / options/ commands that it accepts;
+plz help topics if you want to see the list of possible topics to get help on
+or try a few commands like plz build or plz test if your repo is already set up
+and you'd like to see it in action.
+
+Or see the website (https://please.build) for more information.
+`
+const topicsHelpMessage = `
+The following help topics are available:
+
+%s
+`
+
+var allHelpFiles = []string{"rule_defs.json", "config.json", "misc.json"}
+var allHelpPreambles = []string{rulesPreamble, configPreamble, miscPreamble}
```

Comment:
Instead of having this files-preambles order-dependent mapping here, you could include the preamble in the file itself and nest the help one level deeper. However, I don't know if this makes things any more cohesive, since you probably are still going to have some string constants in this .go file